File: src/core/validators/CommonSimpleType/CommonIntegerMinValueMustNotBeGreaterThanMaxValue.cs
using System;
using MetaEd.Grammar.Antlr;

namespace MetaEd.Core.Validator.CommonSimpleType
{
    public class CommonIntegerMinValueMustNotBeGreaterThanMaxValue : ValidationRuleBase<MetaEdGrammar.CommonIntegerContext>
    {
        public override bool IsValid(MetaEdGrammar.CommonIntegerContext context)
        {
            if (context.minValue() == null || context.maxValue() == null) return true;
            
            // if there are convert exceptions, let it bomb out -- language parser should have handled
            var minValue = Convert.ToInt32(context.minValue().MinValue());
            var maxValue = Convert.ToInt32(context.maxValue().MaxValue());

            return minValue <= maxValue;
        }

        public override string GetFailureMessage(MetaEdGrammar.CommonIntegerContext context)
        {
            return string.Format("Common Integer '{0}' has min value greater than max value.", context.commonIntegerName().GetText());
        }
    }
}

File: src/core/validators/CommonSimpleType/CommonIntegerMinValueMustNotBeGreaterThanMaxValue.js
// @flow
import type SymbolTable from '../SymbolTable';
import { errorRuleBase } from '../ValidationRuleBase';
import { includeRuleBase } from '../ValidationRuleRepository';
import { MetaEdGrammar } from '../../../grammar/gen/MetaEdGrammar';

// eslint-disable-next-line no-unused-vars
export function valid(ruleContext: any, symbolTable: SymbolTable): boolean {
  if (ruleContext.minValue() == null || ruleContext.maxValue() == null) return true;
  const minValue = Number.parseInt(ruleContext.minValue().signed_int().getText(), 10);
  const maxValue = Number.parseInt(ruleContext.maxValue().signed_int().getText(), 10);
  return minValue <= maxValue;
}

// eslint-disable-next-line no-unused-vars
function failureMessage(ruleContext: any, symbolTable: SymbolTable): string {
  return `Common Integer '${ruleContext.commonIntegerName().getText()}' has min value greater than max value.`;
}

const validationRule = errorRuleBase(valid, failureMessage);
// eslint-disable-next-line import/prefer-default-export
export const includeRule = includeRuleBase(MetaEdGrammar.RULE_commonInteger, validationRule);
